src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js

```diff
@@ -0,0 +1,289 @@
+    this.searchContainer.style.display = 'none';
+    this.searchContainer.classList.add(
+      'ecl-select__multiple-dropdown',
+      ...containerClasses
+    );
+    this.selectMultiple.appendChild(this.searchContainer);
+
+    this.search = document.createElement('input');
+    this.search.classList.add('ecl-text-input', 'ecl-text-input--m');
+    this.search.setAttribute('type', 'text');
+    this.search.setAttribute('placeholder', this.textSearch || '');
+    this.searchContainer.appendChild(this.search);
+
+    const selectAll = this.createCheckbox(
+      'all',
+      this.textSelectAll,
+      this.selectMultipleId
+    );
+    selectAll.addEventListener('click', this.handleClickSelectAll);
+    this.searchContainer.appendChild(selectAll);
+
+    if (this.select.options && this.select.options.length > 0) {
+      this.select.options.forEach(option => {
+        const checkbox = this.createCheckbox(
+          option.value,
+          option.text,
+          this.selectMultipleId
+        );
+        checkbox.addEventListener('click', this.handleClickCheckbox);
+        this.searchContainer.appendChild(checkbox);
+      });
+    }
+
+    this.select.parentNode.parentNode.insertBefore(
+      this.selectMultiple,
+      this.select.parentNode.nextSibling
+    );
+
+    this.select.parentNode.classList.add('ecl-select__container--hidden');
+  }
+
+  /**
+   * Destroy component.
+   */
+  destroy() {
+    this.searchContainer.querySelectorAll('input').forEach(input => {
+      input.removeEventListener('click', this.handleClickSelectAll);
+      input.removeEventListener('click', this.handleClickCheckbox);
+    });
+
+    if (this.selectMultiple) {
+      this.selectMultiple.remove();
+    }
+
+    this.select.parentNode.classList.remove('ecl-select__container--hidden');
+  }
+
+  updateInputValue() {
+    this.input.value = Array.from(this.select.options)
+      .filter(option => option.getAttribute('selected'))
+      .map(option => option.text)
+      .join(', ');
+  }
+
+  /**
+   * @param {Event} e
+   */
+  handleClickSelectMultiple(e) {
+    e.preventDefault();
+
+    if (this.searchContainer.style.display === 'none') {
+      this.searchContainer.style.display = 'block';
+    } else {
+      this.searchContainer.style.display = 'none';
+    }
+  }
+
+  /**
+   * @param {Event} e
+   */
+  handleClickCheckbox(e) {
+    e.preventDefault();
+    const checkbox = e.target.parentNode;
+    const input = checkbox.querySelector('input');
+
+    // Toggle values
+    input.checked = !input.checked;
+    this.select.options.forEach(option => {
+      if (option.text === checkbox.getAttribute('data-select-multiple-value')) {
+        if (option.getAttribute('selected')) {
+          option.removeAttribute('selected');
+        } else {
+          option.setAttribute('selected', 'selected');
+        }
+      }
+    });
+
+    this.updateInputValue();
+  }
+
+  /**
+   * @param {Event} e
+   */
+  handleClickSelectAll(e) {
```

Comment:
improvement for the select all: when the select all checkbox is checked, if user uncheck another checkbox, the select all checkbox should also becomes unchecked

Reply:
Nice suggestion, thanks! Implemented.